src/smbclient/_os.py

```diff
@@ -1019,6 +1020,36 @@ def setxattr(path, attribute, value, flags=0, follow_symlinks=True, **kwargs):
         set_info(transaction, ea_info)
 
 
+def liststreams(path, follow_symlinks=True, **kwargs) -> list[str]:
+    """
+    Return a list of the alternative data streams on a path. Listed streams can
+    be opened by appending their name to the original path. An example call for
+    a file with a single extra stream may return:
+
+    ```
+    [":extra_stream:$DATA", "::$DATA"]
+    ```
+
+    :param path: The full UNC path to the file to get the list of streams for.
+    :param follow_symlinks: Whether to follow the symlink at path if encountered.
+    :param kwargs: Common SMB Session arguments for smbclient.
+    :return: List of streams on the file with each entry being a string.
+    """
+
+    raw = SMBRawIO(
```

Comment:
Might need some further investigation to see what is involved here when trying to enumerate the streams on a directory. I'll play around with it a bit and get back to you on that.

Reply:
Looks like it just works so no extra action here needed.